### PR TITLE
199 UI test add a vehicle

### LINKED ISSUE #199: Write a UI Test to Add a Vehicle
## Describe the feature you want the app to have that it currently doesn't
* Write a UI test for adding a `Vehicle` in the `SettingsView`

* Just add it directly to the `BasicCarMaintenanceUITests` file for now, no need to make a new one

### PR DESCRIPTION
**Pull request by am2089**
# What it Does
* Closes #199 
* UI Test for add a vehicle in settings

# How I Tested
Made sure all tests passed before committing

### CODE REVIEW

File: Basic-Car-Maintenance.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -219,8 +221,8 @@
 		FF5D139A2A86C2D500BC9BD6 = {
 			isa = PBXGroup;
 			children = (
-				FFBE79BA2AD0A48C0005524E /* Configurations */,
 				FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */,
+				FFBE79BA2AD0A48C0005524E /* Configurations */,
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/project.pbxproj]: This shouldn't be changed

File: Basic-Car-Maintenance.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -662,6 +666,7 @@
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				ASSETCATALOG_COMPILER_GENERATE_SWIFT_ASSET_SYMBOL_EXTENSIONS = YES;
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
 				CLANG_ANALYZER_NONNULL = YES;
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/project.pbxproj:749]: Do you know what the purpose of this setting is? and why it needs to be set to yes?

File: Basic-Car-Maintenance.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -691,6 +696,7 @@
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
 				COPY_PHASE_STRIP = NO;
+				DEAD_CODE_STRIPPING = YES;
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/project.pbxproj:778]: Do you know what the purpose of this setting is? and why it needs to be set to yes?

File: Basic-Car-Maintenance.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -774,12 +782,14 @@
 		};
 		FF5D13C92A86C2D800BC9BD6 /* Debug */ = {
 			isa = XCBuildConfiguration;
+			baseConfigurationReference = FF098EFA2AB3424E003EC0FE /* Basic-Car-Maintenance.xcconfig */;
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/project.pbxproj:864]: This should not be changed. Re-read the README again to properly set up this project

File: Basic-Car-Maintenance.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -840,6 +851,8 @@
 				"LD_RUNPATH_SEARCH_PATHS[sdk=macosx*]" = "@executable_path/../Frameworks";
 				MACOSX_DEPLOYMENT_TARGET = 13.4;
 				OTHER_LDFLAGS = "-ObjC";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=iphoneos*]" = "com.andrewmuniz.Basic-Car-Maintenance";
+				"PRODUCT_BUNDLE_IDENTIFIER[sdk=macosx*]" = "com.andrewmuniz.Basic-Car-Maintenance";
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance.xcodeproj/project.pbxproj:937]: This should not be changed

File: Basic-Car-Maintenance/Shared/Settings/ViewModels/SettingsViewModel.swift [hunk 6]
```diff
@@ -16,6 +16,23 @@ final class SettingsViewModel {
...
+            case let (contributor1, contributor2) where contributor1 > contributor2:
+                return true
+            case let (contributor1, contributor2) where contributor1 < contributor2:
+                return false
+            default:
+                return contributor1.login < contributor2.login
+            }
+        }
+    }
+
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Settings/ViewModels/SettingsViewModel.swift]: This shouldn't be changed at all, you need to update your fork before making a new branch for new issues

File: Basic-Car-Maintenance/Shared/Settings/Views/ContributorsListView.swift [hunk 7]
```diff
@@ -8,13 +8,12 @@
...
 struct ContributorsListView: View {
-    
     var viewModel: SettingsViewModel
     
     var body: some View {
         List {
-            if let contributors = viewModel.contributors, !contributors.isEmpty {
-                ForEach(contributors) { contributor in
+            if !viewModel.sortedContributors.isEmpty {
+                ForEach(viewModel.sortedContributors) { contributor in
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance/Shared/Settings/Views/ContributorsListView.swift]: This shouldn't be changed

File: Basic-Car-Maintenance-UITests/AddVehicleViewUITest.swift [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        app.launch()
+    }
+    
+    func testAddVehicle() {
+        // Navigate to the AddVehicleView
+        app.buttons["Settings"].tap()
+        app.buttons["Add Vehicle"].tap()
+        
+        // Fill in valid vehicle information
+        let nameTextField = app.textFields["Vehicle Name"]
```
Comment (mikaelacaron) [inline Basic-Car-Maintenance-UITests/AddVehicleViewUITest.swift]: This test failed for me:
> Failed to tap "Vehicle Name" TextField: No matches found for Descendants matching type TextField from input

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
Good work! Note I did specifically say in the issue to
> Just add it directly to the `BasicCarMaintenanceUITests` file for now, no need to make a new one

Which this has caused a merge conflict that needs to be resolved, also I don't think you updated your fork, or what branch you worked on. There's changes in the `SettingsViewModel` that shouldn't be there.

Also note several files changed that I wouldn't expect, in the `xcschemes/WidgetExtension.xcscheme` and
`/xcschemes/Basic-Car-Maintenance-WidgetExtension.xcscheme`

Address these issues and click the re-review button when you're ready, comment on the thread if you have any questions

Re-review button

**Comment by am2089** [comment]
Hey, so i made sure my my fork was up to date now, and moved the UITest into the file you wanted. I think the pbx file was old before you made the changes and it was just a commit that was still there that got pushed with the UITest I made. I could be doing something wrong lol but I double checked that I am up to date now and just pushed the changed where I moved the code.

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
@am2089 there's still several changes that are there but should be and from looking at GitHub it's still saying your branch is **not** up to date with the latest in `dev`

**Comment by am2089** [comment]
Im just curious, should I just re-fork the project so it can be absolutely up to date? and just add the test I wrote, and create a new pull request so everything will be on a clean slate?

**Comment by mikaelacaron** [comment]
> Im just curious, should I just re-fork the project so it can be absolutely up to date? and just add the test I wrote, and create a new pull request so everything will be on a clean slate?

No. You need to update the `dev` branch of you fork by clicking the sync button in GitHub, and then merge those changes into you branch. I did a livestream about this the other day. Start around `1:20:44` https://www.youtube.com/live/-rYYKXrCyro?si=J2IQNjJyqa6wiwrQ

**Comment by am2089** [comment]
So I did that before I even made the fork and the branch. 
Screenshot 2023-10-22 at 9 12 12 PM

**Comment by am2089** [comment]
Actually i see  whats wrong my branch isn't updated and is 7 commits ahead, 54 commits behind. So should I delete the seven commits, update the branch and re commit the changes?

**Comment by mikaelacaron** [comment]
No don't need to delete any commits. You only need to update your branch, (merging the changes from `dev` into your branch `199-UITest-AddAVehicle`

**Comment by am2089** [comment]
So I fixed the pbx file in vscode, saved it, and committed it. Is there anything else I need to do? Or is it what I did not showing? also should i start a new pr, or did you close it because it was accepted?

**Comment by mikaelacaron** [comment]
> So I fixed the pbx file in vscode, saved it, and committed it. Is there anything else I need to do? Or is it what I did not showing? also should i start a new pr, or did you close it because it was accepted?

You fix the code on your branch, and then push the changes to GitHub. There was no need to close the PR.
I didn't close the PR, you did.

But you can either re-open this one, or make a new one, it doesn't matter as long as it's been updated as shown in my livestream that I linked earlier

**Comment by mikaelacaron** [comment]
@am2089 There's still a bunch of changes that I noted that haven't been fixed.
I think the best thing to do would be to copy the contents of the `BasicCarMaintenanceUITests.swift` file for what you've changed, and start a new branch on your fork from `dev` (because it will be updated)

Apply your changes, and then make a new PR

**Review by mikaelacaron — CHANGES_REQUESTED** [review]
And the only file that should be changed would be the `BasicCarMaintenanceUITests.swift`, _unless_ you can explain why any Xcode settings need to be changed

**Comment by am2089** [comment]
So I'm sort of confused about what to do. I understand you pointed out all those commits, but I only want to commit the commits I made on Oct 22. How do I get rid of the rest? Sorry wrapping my head around git and Git Hub when a file gets messed up is a task. Also on a side note will you be continuing this project passed October?

**Comment by mikaelacaron** [comment]
> How do I get rid of the rest? Sorry wrapping my head around git and Git Hub when a file gets messed up is a task. Also on a side note will you be continuing this project passed October?


One thing to do is just make a new branch and re commit everything, that's the simplest!

otherwise, you need to use a text editor and manually remove the stuff that shouldn't be there. Actually "removing commits" is possible, but significantly more difficult than you would expect.

Also yes! I'll be continuing this project after October! I need to add a paywall and put features behind the paywall. I'm hoping to ship within the first few weeks of November! This project will still be open for people to contribute, but I imagine it'll significantly slow down after October is over. You're welcome to still contribute to it!


Easiest way to fix this:
* Make sure your fork is up to date
* Checkout a new branch from `dev`
* Move the one change you had (everything in the `BasicCarMaintenanceUITests.swift` file)
   * Literally copy and paste what your change is into the new branch and run it
* Commit that
* Change the PR at the top to that new branch (or create a new PR)

**Pull request closed without merging**